compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUVectorLowering.cpp

```diff
@@ -183,6 +184,288 @@ struct SetMulAddFMF final : OpRewritePattern<vector::MultiDimReductionOp> {
   }
 };
 
+struct ContractToChainFMA final : OpRewritePattern<vector::ContractionOp> {
```

Comment:
Can you add a comment with an example of a small contract and its expansions? Also explain why we want this rewrite.